# Exercises and Solutions for INF281 Basics of Bioinformatics Sequence Analysis

[![Build Status](https://travis-ci.org/takayasaito/INF281-Exercises.svg?branch=master)](https://travis-ci.org/takayasaito/INF281-Exercises)

This repository contains the sources files of the exercises and solutions for INF281 Basics of bioinformatics sequence analysis. The generated PDFs are available on the [Release page](https://github.com/takayasaito/INF281-Exercises/releases).

INF281 Lecture Notes are availabe [here](https://github.com/takayasaito/INF281-Lecture-Notes).

## Gropup exercises
| Exercise No | Section No | Section Name                          |
|-------------|------------|---------------------------------------|
| 01          | 01-1       | Introduction to Molecular Biology     |
|             | 02-1       | Global pairwise alignment             |
|             | 02-2       | Alignment by brute-force              |
|             | 02-3       | Table representation of alignment     |
| 02          | 02-4       | Global alignment with DP              |
|             | 02-5       | Backtracking                          |
|             | 03-1       | Usage of score matrices               |
| 03          | 03-3       | General gap penalties                 |
|             | 03-4       | Affine gap penalties                  |
|             | 03-5       | Sequence distance                     |
| 04          | 04-2       | Dot matrix                            |
|             | 04-3       | Local alignment with DP               |
|             | 05-3       | N-gram based search                   |
|             | 05-4       | Lookup table of matching n-grams      |
| 05          | 05-5       | Finite-state machine with n-grams     |
|             | 06-2       | Evaluation of global alignment        |
|             | 06-3       | Evaluation of local alignment         |
| 06          | 06-4       | Evaluation of database search         |
|             | 06-5       | Bit score and e-value                 |
|             | 07-1       | Evaluation of binary classifiers      |
|             | 07-2       | Confusion matrix                      |
|             | 07-3       | Basic evaluation measures             |
| 07          | 07-4       | Threshold free measures               |
|             | 08-1       | Multiple alignment                    |
|             | 09-1       | Phylogenetic tree                     |
|             | 09-3       | UPGMA                                 |
| 08          | 09-4       | Maximum parsimony                     |
|             | 09-5       | Maximum likelihood                    |
|             | 10-1       | Progressive alignment                 |
| 09          | 10-2       | Alignment clustering                  |
|             | 10-3       | Aligning methods                      |
| 10          | 11-1       | Scoring schemes for protein sequences |
|             | 11-2       | PAM - accepted mutations              |
| 11          | 11-3       | PAM - substitution matrix             |
|             | 11-4       | BLOSUM                                |
|             | 12-2       | Position weight matrix                |
|             | 12-3       | Sequence profiles                     |
| 12          | 12-3       | Sequence profiles                     |
|             | 13-1       | Hidden Markov model                   |
|             | 13-2       | Viterbi algorithm                     |
|             | 13-3       | HMM profile                           |
| 13          | 14-1       | Sequence patterns                     |
|             | 14-3       | Pattern discovery                     |
|             | 15-1       | RNA structure                         |
|             | 15-2       | Nussinov algorithm                    |
